deploy-cloud-run/action.yml

```diff
@@ -45,11 +63,134 @@ runs:
+            local path=$(echo "$url" | sed -E 's|https?://[^/]+||')
+            # If path is empty, set to "/"
+            if [[ -z "$path" ]]; then
+                path="/"
+            fi
+            echo "url_path=$path" >> "${GITHUB_OUTPUT}"
+        }
+        extract_path "${{ inputs.test_url }}"
+      shell: bash
+    - name: Get current Cloud Run service revision
+      id: get_revision
+      if: ${{ inputs.automatic_rollback_enabled == 'true' || inputs.blue_green_enabled == 'true' }}
+      shell: bash
+      run: |
+        echo "Getting current cloudrun revision..."
+        REV=$( \
+          gcloud run services describe "${{ inputs.service }}" \
+            --region "${{ inputs.region }}" \
+            --format="value(status.latestReadyRevisionName)" \
+        )
+        echo "Current revision: $REV"
+        echo "revision=$REV" >> "$GITHUB_OUTPUT"
+    - name: Logging - Deploy to Cloud Run
+      run: echo "Deploying to Cloud Run..."
+      if: ${{ inputs.blue_green_enabled == 'false' }}
+      shell: bash
     - name: Deploy to Cloud Run
       uses: google-github-actions/deploy-cloudrun@v2
+      if: ${{ inputs.blue_green_enabled == 'false' }}
+      id: deploy
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        revision_traffic: ${{ inputs.to_latest == 'true' && 'LATEST=100' || '' }}
+        skip_default_labels: true
+    - name: Run HTTP health check
+      continue-on-error: true
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'false' && inputs.test_url != '' }}
+      id: test_live
+      with:
+        url: ${{ inputs.test_url }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
+    - name: Logging - Deploy to Cloud Run Blue
+      run: echo "Deploying ${{ inputs.image_internal || inputs.image }} to ${{ inputs.service }} blue..."
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      shell: bash
+    - name: Deploy to Cloud Run Blue
+      uses: google-github-actions/deploy-cloudrun@v2
+      id: deploy_blue
+      if: ${{ inputs.blue_green_enabled == 'true' }}
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        tag: blue-release
+        no_traffic: true
+        skip_default_labels: true
+    - name: Run HTTP health check on blue
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'true' && inputs.test_url != ''}}
+      id: test_blue
+      with:
+        url: ${{ steps.deploy_blue.outputs.url }}${{ steps.parse_url.outputs.url_path }}
+        method: 'GET'
+        ignoreSsl: true
+        ignoreStatusCodes: ${{ steps.http_responses.outputs.codes }}
```

Comment:
@rot1024 I use cloud run URL for blue testing since blue will receive 0% traffic from LB during this phase. So the only viable test is by hitting the blue endpoint directly using tag---cloudrunurl